### ISSUE TITLE
Unexpected templating type error occurred on (<template_content>) TypeError: sequence item 2: expected str instance, int found

### ISSUE THREAD

**Issue by simonsunx** [willshersystems/ansible-sshd#188]
Hi.

I've got error
> Unexpected templating type error occurred on (<template_content>) TypeError: sequence item 2: expected str instance, int found

 while running `ansible-playbook`.

So I changed var types:

- before:
```yaml
sshd:
  PermitRootLogin: no
  ClientAliveCountMax: 99999
  ClientAliveInterval: 20
  RSAAuthentication: yes
  PasswordAuthentication: no
...
```

- after:
```yaml
sshd:
  PermitRootLogin: 'no'
  ClientAliveCountMax: '99999'
  ClientAliveInterval: '20'
  RSAAuthentication: 'yes'
  PasswordAuthentication: 'no'
...
```

After editing error has gone.

Ansible version:
```bash
$ ansible --version
ansible [core 2.13.0]
  config file = /etc/ansible.cfg
  configured module search path = ['/home/login/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.10/site-packages/ansible
  ansible collection location = /home/login/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.10.4 (main, Mar 23 2022, 23:05:40) [GCC 11.2.0]
  jinja version = 3.0.3
  libyaml = True
```

I hope this experience will help someone.

**Comment by Jakuje** [comment]
Thank you for the report. Could this be something new in the ansible 2.13? We are running various ansible versions for some time (currently 2.12 in github actions) and they do not demonstrate this issue.

**Comment by richm** [comment]
> Thank you for the report. Could this be something new in the ansible 2.13? We are running various ansible versions for some time (currently 2.12 in github actions) and they do not demonstrate this issue.

Yep - sounds like an issue with ansible-core 2.13

**Comment by Jakuje** [comment]
Can you clarify if this is an issue *of* ansible-core 2.13 or *of* our role with ansilbe-core-2.13?

**Comment by richm** [comment]
@nhosoi This looks like a 2.13 porting issue - might be an issue with jinja 3 as well - maybe ansible-core 2.13 or jinja 3 is more type strict?  I'm guessing we may have to do some sort of variable type coercion in the template code, but I'm not sure where.

**Comment by richm** [comment]
This looks like a bug in jinja 3.0.3.  If I run with ansible-core-2.13 and jinja 3.0.3 I see the error.  If I use jinja latest (3.1.2) I do not see the error.

**Comment by nhosoi** [comment]
I got the same results with "jinja version = 3.1.2".

**Comment by richm** [comment]
The role uses macros, and for some reason, integers are not rendered as strings correctly inside macros.  The following playbook is an easy reproducer:

Steps:
1) python -mvenv .venv
2) . .venv/bin/activate
3) pip install ansible-core==2.13.* jinja2==3.0.*
4) ansible-playbook -vv jinja30-error.yml

Using this playbook:
```yaml
- hosts: localhost
  gather_facts: false
  vars:
    mydict:
      a: a
      b: 1
  tasks:
    - name: show error
      debug:
        msg: |
          {% macro doerror(key,value) %}
          {{ key }} {{ value }}
          {% endmacro %}
          {{ doerror('b', mydict['b']) }}
```
you get this error
```
Unexpected templating type error occurred on ({% macro doerror(key,value) %}
{{ key }} {{ value }}
{% endmacro %}
{{ doerror('b', mydict['b']) }}
): sequence item 2: expected str instance, int found
```

This does not have any errors when using jinja 2.x or jinja 3.1.x.

The workaround is to use string values e.g. use `Port: "2200"` instead of `Port: 2200` - at least, that seems to work in my limited testing.

You can fix this by using {{ value | string }}, but this seems like a bug in jinja 3.0.x since it works correctly without the "| string" on 2.x and 3.1.x

**Comment by richm** [comment]
ping - just wanted to check in and see if everyone affected has been able to use jinja 3.1.x - I know that the RHEL 8.7/9.1 ansible-core 2.13 package will bundle jinja 3.1.x

**Comment by Jakuje** [comment]
This was fixed with #197 and will be in the next release
